lib/model/app_model/pay_invoice.dart

```diff
@@ -1,19 +1,64 @@
 import 'package:cln_common/cln_common.dart';
+import 'package:clnapp/model/app_model/app_utils.dart';
 
 class AppPayInvoice {
-  Map<String, dynamic> payResponse;
+  AppPay payResponse;
 
   AppPayInvoice({required this.payResponse});
 
   //TODO: how manage this? looks incompleted.
   factory AppPayInvoice.fromJSON(Map<String, dynamic> json,
       {bool snackCase = false}) {
-    var payResponse = json;
-    LogManager.getInstance.debug("$json");
-    if (payResponse.isNotEmpty) {
-      return AppPayInvoice(payResponse: payResponse);
-    } else {
-      return AppPayInvoice(payResponse: json);
-    }
+    LogManager.getInstance.debug("RESPONSE HERE : $json");
+    var payResponse = AppPay.fromJSON(json);
+    return AppPayInvoice(payResponse: payResponse);
+  }
+}
+
+class AppPay {
+  final String destination;
+
+  final String paymentHash;
+
+  final String createdAt;
+
+  // amount that the recipient received
+  final String amountMSAT;
+
+  final String paymentPreimage;
+
+  final String status;
+
+  // amount that we sent including the fees.
+  final String amountSentMSAT;
+
+  AppPay({
+    required this.destination,
+    required this.paymentPreimage,
+    required this.createdAt,
+    required this.status,
+    required this.amountMSAT,
+    required this.paymentHash,
+    required this.amountSentMSAT,
+  });
+
+  factory AppPay.fromJSON(Map<String, dynamic> json, {bool snackCase = false}) {
+    var destination = json.withKey("destination", snackCase: snackCase);
+    var paymentPreimage =
+        json.parseMsat(key: "payment_preimage", snackCase: true);
```

Comment:
this should be withKey right?

Reply:
Yes, it should be. Thank you for letting me know. I'll fix this in the follow up PR.